pkg/blobinfocache/boltdb/boltdb.go

```diff
@@ -220,6 +226,30 @@ func (bdc *cache) RecordDigestUncompressedPair(anyDigest digest.Digest, uncompre
 	}) // FIXME? Log error (but throttle the log volume on repeated accesses)?
 }
 
+// RecordDigestCompressorName records that the blob with digest anyDigest was compressed with the specified
+// compressor, or is blobinfocache.Uncompressed.
+// WARNING: Only call this for LOCALLY VERIFIED data; don’t record a digest pair just because some remote author claims so (e.g.
+// because a manifest/config pair exists); otherwise the cache could be poisoned and allow substituting unexpected blobs.
```

Comment:
Right, adding this:
```
WARNING: Only call this with LOCALLY VERIFIED data; don’t record a compressor for a
digest just because some remote author claims so (e.g. because a manifest says so);
otherwise the cache could be poisoned and cause us to make incorrect edits to type
information in a manifest.
```